### PR TITLE
integration_tests/Makefile: do not use docker pull

### PR DESCRIPTION
**Pull request by fredericlepied**
Depends-On: https://github.com/wazo-platform/xivo-test-helpers/pull/25
Depends-On: https://github.com/wazo-platform/sf-jobs/pull/42

### CODE REVIEW

**Comment by fredericlepied** [comment]
recheck

File: integration_tests/Makefile [hunk 1]
```diff
@@ -1,23 +1,16 @@
...
-	docker pull munkyboy/fakesmtp
-	docker pull nickstenning/slapd
-	docker pull rabbitmq
-	docker pull wazopbx/postgres-test
-	docker pull wazopbx/wait
-
 test-image:
-	docker build --no-cache -t wazopbx/wazo-auth ..
-	docker build --no-cache -t wazopbx/wazo-auth-tests -f Dockerfile-auth-test ..
+	docker image inspect --format='{{ .Id }}' wazopbx/wazo-auth || docker build --no-cache -t wazopbx/wazo-auth ..
```
Comment (sileht) [inline integration_tests/Makefile]: This will make the image not rebuilt when we run tox -e intergration locally?

Comment (fredericlepied) [reply integration_tests/Makefile]: yes that's the drawback. We could try to avoid this in sf-jobs in another way.

Comment (fblackburn1) [reply integration_tests/Makefile]: Please don't change this behavior. There are enough of corner case with integration tests, we need to be confident that if we do a `make test-setup` we start with a very clean setup and if we have an error this is the code fault, not the environment.

Comment (fredericlepied) [reply integration_tests/Makefile]: yes I have found a solution for this. Don't worry ;-)

Comment (fblackburn1) [reply integration_tests/Makefile]: Don't see the other commits, please mark the discussion as solved with the decision taken

Comment (fredericlepied) [reply integration_tests/Makefile]: The need for cache us solved by avoiding the current PR in https://github.com/wazo-platform/sf-jobs/pull/41

File: integration_tests/Makefile [hunk 2]
```diff
@@ -1,20 +1,11 @@
...
 test-db:
 	docker build -f ../contribs/docker/Dockerfile-db -t wazopbx/wazo-auth-db ..
 
-test-prerequisite:
-	docker pull munkyboy/fakesmtp
-	docker pull nickstenning/slapd
-	docker pull rabbitmq
-	docker pull wazopbx/postgres-test
-	docker pull wazopbx/wait
-
```
Comment (fblackburn1) [inline integration_tests/Makefile]: Don't remove docker pull
`docker-compose up` will not pull new image.
Why we need it: https://github.com/wazo-platform/wazo-amid/pull/11#discussion_r363891392

Comment (fredericlepied) [reply integration_tests/Makefile]: Why do you say it will not pull a new image as the doc is stating the opposite except if you use --no-recreate: https://docs.docker.com/compose/reference/up/

And this is needed to let docker-comoose take a local fresh build instead of forcing the download from the remote registry. We need it to have docker images built from dependent PRs working.

Comment (fblackburn1) [reply integration_tests/Makefile]: the documentation do not state of anything about pulling the image from registry. The only thing it mention is if a CONTAINER is started without the last image on the host, then it will be recreated.
You can try the following scenario
Given new host without any image
Given `docker-compose up` <-- new image will be pulled from registry
When I wait until new image is available on docker registry
When `docker-compose up`
Then nothing is changed, because image already exist on host

The `docker-compose up` command do not pull image from registry if it already exist on host.

I understand problems with zuul and local image, but I not agree with this solution since it can cause other problems for developer environment.

**Comment by fredericlepied** [comment]
recheck

**Comment by fredericlepied** [comment]
recheck

**Comment by fredericlepied** [comment]
recheck

**Comment by fredericlepied** [comment]
recheck

**Comment by fredericlepied** [comment]
recheck

File: integration_tests/Makefile (hunk 2 above)
Comment (fredericlepied) [reply integration_tests/Makefile]: Ok I tried to make it work for both cases by using docker-compose pull automatically except in CI in this: https://github.com/wazo-platform/xivo-test-helpers/pull/25/files#

Let me know if this is ok for you.

Comment (fblackburn1) [reply integration_tests/Makefile]: Yes it is a solution that I would like to have time to implement, thanks !! :+1:

File: integration_tests/test-requirements-for-tox.txt [hunk 3]
```diff
@@ -4,7 +4,7 @@ kombu
 mock
 pytest
 openapi-spec-validator
-pyhamcrest
+pyhamcrest!=1.10
```
Comment (fblackburn1) [inline integration_tests/test-requirements-for-tox.txt]: The error was fixed by https://github.com/wazo-platform/xivo-test-helpers/pull/24
No need to add version dependency

Comment (fblackburn1) [reply integration_tests/test-requirements-for-tox.txt]: fixed: revert this change

**Comment by fblackburn1** [comment]
There are any reason why moving `.PHONY` on bottom of the file?
Is it a Makefile standard or guideline?
Personally I don't care, it's just that all our Makefile have the `PHONY` on top of the file, so I would like to know which standard we should use?

**Comment by fredericlepied** [comment]
> There are any reason why moving `.PHONY` on bottom of the file?
> Is it a Makefile standard or guideline?
> Personally I don't care, it's just that all our Makefile have the `PHONY` on top of the file, so I would like to know which standard we should use?

Yes that's the usual way to have only one .PHONY at the end of the file.

**Comment by fblackburn1** [comment]
@fredericlepied do you miss the mergeit label or you don't want to mergeit for now?
